lib/block-supports/duotone.php

```diff
@@ -445,15 +442,44 @@ function gutenberg_render_duotone_support( $block_content, $block ) {
 		return $block_content;
 	}
 
-	$colors          = $block['attrs']['style']['color']['duotone'];
-	$filter_key      = is_array( $colors ) ? implode( '-', $colors ) : $colors;
-	$filter_preset   = array(
-		'slug'   => wp_unique_id( sanitize_key( $filter_key . '-' ) ),
-		'colors' => $colors,
-	);
-	$filter_property = gutenberg_get_duotone_filter_property( $filter_preset );
-	$filter_id       = gutenberg_get_duotone_filter_id( $filter_preset );
+	// Possible values for duotone attribute:
+	// 1. array('#000000', '#ffffff').
+	// 2. 'preset-slug'.
+	// 3. 'unset' (remove filter).
+	$duotone_attr = $block['attrs']['style']['color']['duotone'];
+
+	$is_duotone_colors_array = is_array( $duotone_attr ) && count( $duotone_attr ) === 2;
```

Comment:
What situation would cause the `count()` to be something other than 2?

Reply:
OK - I see an instance of it using [three colors in the `sherbert.json` for Twenty Twenty Three](https://github.com/WordPress/twentytwentythree/blob/trunk/styles/sherbet.json#L9-L13). I'm not sure yet how it's treated differently though if it's using two or three or more colors.

If it's not handled differently, I'd say remove the `count( $duotone_attr ) === 2` check. If it is handled differently, I'd argue we should rename `$is_duotone_colors_array` to something else since if it has three colors, it's still an array, so the current name would be misleading.

Reply:
> I see an instance of it using [three colors in the sherbert.json for Twenty Twenty Three](https://github.com/WordPress/twentytwentythree/blob/trunk/styles/sherbet.json#L9-L13)

Nice! I didn't realize that made it into TT3.

Reply:
Thanks for the context! So, remove the `count` check here, and then use `$is_duotone_colors_array` in place of the [`$filter_key      = is_array( $duotone_attr ) ? …` check](https://github.com/WordPress/gutenberg/pull/48318#discussion_r1114800819) later should work well. I'll put a little commit in.

Reply:
Thanks for the context and fix 👍